lib/thread/no_os/semaphore.cpp

```diff
@@ -69,16 +86,26 @@ void ecl::binary_semaphore::wait()
 
 bool ecl::binary_semaphore::try_wait(std::chrono::milliseconds ms)
 {
+
+    bool ex;
+
+#ifdef USE_SYSTMR
```

Comment:
I believe it is possible to omit the check for `USE_SYSTMR` here. The `wait_for` implementation takes care of both cases anyway. That is, of course, if `ecl::clk()` is provided on all targets, does it?